backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inventory/interceptor/M_InventoryLine.java

```diff
@@ -64,6 +68,13 @@ public void registerCallout()
 		Services.get(IProgramaticCalloutProvider.class).registerAnnotatedCallout(this);
 	}
 
+	@CalloutMethod(columnNames = org.compiere.model.I_M_InventoryLine.COLUMNNAME_M_Product_ID)
+	public void setUOM(@NonNull final org.compiere.model.I_M_InventoryLine inventoryLine)
+	{
+		final UomId stockUOMId = inventoryLine.getM_Product_ID() > 0 ? productBL.getStockUOMId(ProductId.ofRepoId(inventoryLine.getM_Product_ID())) : UomId.ofRepoIdOrNull(0);
+		inventoryLine.setC_UOM_ID(stockUOMId != null ? stockUOMId.getRepoId() : 0);
```

Comment:
```suggestion
		final UomId stockUOMId = inventoryLine.getM_Product_ID() > 0 ? productBL.getStockUOMId(inventoryLine.getM_Product_ID()) : null;
		inventoryLine.setC_UOM_ID(UomId.toRepoId(stockUOMId));
```

simplified the method call